scene/2d/physics/character_body_2d.cpp

```diff
@@ -352,14 +352,12 @@ void CharacterBody2D::_apply_floor_snap(bool p_wall_as_floor) {
 			floor_normal = result.collision_normal;
 			_set_platform_data(result);
 
-			if (floor_stop_on_slope) {
-				// move and collide may stray the object a bit because of pre un-stucking,
-				// so only ensure that motion happens on floor direction in this case.
-				if (result.travel.length() > margin) {
-					result.travel = up_direction * up_direction.dot(result.travel);
-				} else {
-					result.travel = Vector2();
-				}
+			// move and collide may stray the object a bit because of pre un-stucking,
+			// so only ensure that motion happens on floor direction in this case.
```

Comment:
```suggestion
			// Ensure that we only move the body along the up axis, because
			// move_and_collide may stray the object a bit when getting it unstuck.
```